File: w5d4-IntroJavaScript-master/array/array.js
var uniq = function(arr) {
  var result = [];
  for (var i = 0; i < arr.length; i++) {
    if (result.indexOf(arr[i]) === -1) {
      result.push(arr[i]);
    }
  }
  return result;
};


var twoSum = function(arr) {
  var outerIdx = 0;
  var innerIdx = 1;
  var result = [];

  while (outerIdx < arr.length - 1) {
    while (innerIdx < arr.length) {
      if (arr[outerIdx] + arr[innerIdx] === 0) {
        result.push([outerIdx, innerIdx]);
      }
      innerIdx++;
    }

    outerIdx++;
    innerIdx = outerIdx + 1;
  }

  return result;
};

var myTranspose = function(arr) {
  var result = new Array(arr.size);

  for (var i = 0; i < arr.length; i++) {
    result[i] = new Array(arr.size);
  }

  for (var i = 0; i < arr.length; i++) {
    for (var j = 0; j < arr[i].length; j++) {
      result[i][j] = arr[j][i];
    }
  }
  console.log(result);
  return result;
};

File: w2d5-master/array/lib/array.rb
class Array

  def my_uniq
    arr = []
    self.each do |el|
      arr << el if !arr.include?(el)
    end
    arr
  end

  def two_sum
    array = []
    i = 0
    until i == self.size - 1
      j = i + 1
      until j == self.size
        if self[i] + self[j] == 0
          array << [i, j]
        end
        j += 1
      end
      i += 1
    end
    array
  end

  def median
    array = self.sort
    if array.size % 2 == 0
      return (array[array.size/2] + array[array.size/2 - 1])/2
    else
      return array[array.size/2]
    end
  end

  def my_transpose
    result = Array.new(self.size) { Array.new(self.size) }
    (0..self.size-1).each do |i|
      (0..self.size-1).each do |j|
        result[j][i] = self[i][j]
      end
    end
    result
  end

end
